clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1053,12 +1053,20 @@ public void testFetchStableOffsetThrowInCommitted(GroupProtocol groupProtocol) {
         assertThrows(UnsupportedVersionException.class, () -> setupThrowableConsumer(groupProtocol).committed(Collections.singleton(tp0)));
     }
 
-    // TODO: this test triggers a bug with the CONSUMER group protocol implementation.
-    //       The bug will be investigated and fixed so this test can use both group protocols.
     @ParameterizedTest
-    @EnumSource(value = GroupProtocol.class, names = "CLASSIC")
-    public void testFetchStableOffsetThrowInPoll(GroupProtocol groupProtocol) {
-        assertThrows(UnsupportedVersionException.class, () -> setupThrowableConsumer(groupProtocol).poll(Duration.ZERO));
+    @EnumSource(GroupProtocol.class)
+    public void testFetchStableOffsetThrowInPoll(GroupProtocol groupProtocol) throws InterruptedException {
+        setupThrowableConsumer(groupProtocol);
+        TestUtils.waitForCondition(() -> {
```

Comment:
this change still has me thinking. This test is about a single call to `poll(ZERO)`, that is expected to throw an exception, but interesting fact is that the exception is generated when building the request [here](https://github.com/apache/kafka/blob/70dd577286de31ef20dc4f198e95f9b9e4479b47/clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java#L146) (it does not require the actual send or response). So I wonder if the async consumer should somehow ensure that when poll returns (even with low time), it has allowed for at least one run of the background thread runOnce?

Reply:
Hi @lianetm, the `ApplicationEventProcessor#process(PollEvent)` only resets timer for some request managers. Even if we use `ApplicationEventHandler#addAndGet`, we still can't guarantee `runOnce` is executed. We may need `ConsumerNetworkThread` to send a `BackgroundEvent` to make sure the `runOnce` is happened. However, this approach may make the process more complex. If we want to make this, we probably can create another Jira to handle this. WDYT?

Reply:
Agree that it's better to discuss separately. Definitely we could consider to align the behaviour of the 2 consumers a bit more regarding poll with low timeouts and the guarantees of requests sent, but there are tradeoffs to consider as @chia7712 pointed out on his [comment](https://issues.apache.org/jira/browse/KAFKA-16792?focusedCommentId=17878298&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17878298) on the Jira (totally agree on the trade-offs).